core/include/cubos/core/al/audio_context.hpp

```diff
@@ -0,0 +1,200 @@
+/// @file
+/// @brief Class @ref cubos::core::al::AudioContext and related types.
+/// @ingroup core-al
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <glm/glm.hpp>
+
+#include <cubos/core/api.hpp>
+
+namespace cubos::core::al
+{
+    namespace impl
+    {
+        class Buffer;
+        class Source;
+        class Listener;
+        class AudioDevice;
+    } // namespace impl
+
+    /// @brief Handle to an audio buffer.
+    /// @see impl::Buffer - audio buffer interface.
+    /// @see AudioContext::createBuffer()
+    /// @ingroup core-al
+    using Buffer = std::shared_ptr<impl::Buffer>;
+
+    /// @brief Handle to an audio source.
+    /// @see impl::Source - audio source interface.
+    /// @see AudioDevice::createSource()
+    /// @ingroup core-al
+    using Source = std::shared_ptr<impl::Source>;
+
+    /// @brief Handle to an audio listener.
+    /// @see impl::Listener - audio listener interface.
+    /// @see AudioDevice::createListener()
+    /// @ingroup core-al
+    using Listener = std::shared_ptr<impl::Listener>;
+
+    /// @brief Handle to an audio device.
+    /// @see impl::AudioDevice - audio device interface.
+    /// @see AudioContext::createDevice()
+    /// @ingroup core-al
+    using AudioDevice = std::shared_ptr<impl::AudioDevice>;
+
+    /// @brief Audio context that contains audio devices;
+    class CUBOS_CORE_API AudioContext
+    {
+    public:
+        AudioContext() = default;
+        virtual ~AudioContext() = default;
+
+        /// @brief Creates an audio context.
+        /// @return AudioContext, or nullptr on failure.
+        static std::shared_ptr<AudioContext> create();
+
+        /// @brief Enumerates the available devices.
+        /// @param[out] devices Vector to fill with the available devices.
+        virtual void enumerateDevices(std::vector<std::string>& devices) = 0;
+
+        /// @brief Creates a new audio device
+        /// @param specifier The specifier of the audio device, used to identify it
+        /// @param listenerCount The number of audio listener to be created by the device's engine.
+        /// @return Handle of the new device
+        virtual AudioDevice createDevice(unsigned int listenerCount,
+                                         const std::string& specifier = "MiniaudioDevice") = 0;
```

Comment:
There shouldn't be a reference to Miniaudio here, this is supposed to be abstract.
You could use an empty specifier as a default value, and then handle it on the miniaudio device as necessary.